max_auto_associations_e2e_test.go

```diff
@@ -0,0 +1,598 @@
+}
+
+func TestUnlimitedMaxAutoAssociationsAllowsToTransferFromFungibleTokens(t *testing.T) {
+	t.Parallel()
+	env := NewIntegrationTestEnv(t)
+
+	// create spender account which will be approved to spend
+	spender, spenderKey := createAccountHelper(t, &env, 10)
+
+	// create token1 with 1 mil supply
+	tokenID1 := createFungibleTokenHelper(3, t, &env)
+
+	// create token2 with 1 mil supply
+	tokenID2 := createFungibleTokenHelper(3, t, &env)
+
+	// account create with unlimited max auto associations
+	accountID1, _ := createAccountHelper(t, &env, -1)
+	// account update with unlimited max auto associations
+	accountID2, newKey := createAccountHelper(t, &env, 100)
+	accountUpdateFrozen, err := NewAccountUpdateTransaction().
+		SetMaxAutomaticTokenAssociations(-1).
+		SetAccountID(accountID2).
+		FreezeWith(env.Client)
+	require.NoError(t, err)
+
+	accountUpdate, err := accountUpdateFrozen.Sign(newKey).Execute(env.Client)
+	require.NoError(t, err)
+
+	_, err = accountUpdate.SetValidateStatus(true).GetReceipt(env.Client)
+	require.NoError(t, err)
+
+	// approve the spender
+	approve, err := NewAccountAllowanceApproveTransaction().
+		AddTokenApproval(tokenID1, spender, 2000).
+		AddTokenApproval(tokenID2, spender, 2000).
+		Execute(env.Client)
+	require.NoError(t, err)
+
+	_, err = approve.SetValidateStatus(true).GetReceipt(env.Client)
+	require.NoError(t, err)
+
+	// transferFrom to both accounts some token1 tokens
+	tokenTransferTransactionFrozen, err := NewTransferTransaction().
+		AddTokenTransfer(tokenID1, env.Client.GetOperatorAccountID(), -1000).
+		AddTokenTransfer(tokenID1, accountID1, 1000).
+		AddTokenTransfer(tokenID1, env.Client.GetOperatorAccountID(), -1000).
+		AddTokenTransfer(tokenID1, accountID2, 1000).
+		FreezeWith(env.Client)
+	require.NoError(t, err)
+
+	tokenTransferTransaction, err := tokenTransferTransactionFrozen.Sign(spenderKey).Execute(env.Client)
+
+	_, err = tokenTransferTransaction.SetValidateStatus(true).GetReceipt(env.Client)
+	require.NoError(t, err)
+
+	// transferFrom token2 to all some tokens
+	tokenTransferTransactionFrozen, err = NewTransferTransaction().
+		AddTokenTransfer(tokenID2, env.Client.GetOperatorAccountID(), -1000).
+		AddTokenTransfer(tokenID2, accountID1, 1000).
+		AddTokenTransfer(tokenID2, env.Client.GetOperatorAccountID(), -1000).
+		AddTokenTransfer(tokenID2, accountID2, 1000).
+		FreezeWith(env.Client)
+	require.NoError(t, err)
+
+	tokenTransferTransaction, err = tokenTransferTransactionFrozen.Sign(spenderKey).Execute(env.Client)
+
+	_, err = tokenTransferTransaction.SetValidateStatus(true).GetReceipt(env.Client)
+	require.NoError(t, err)
+}
+
+func TestUnlimitedMaxAutoAssociationsAllowsToTransferNFTs(t *testing.T) {
+	t.Parallel()
+	env := NewIntegrationTestEnv(t)
+
+	// create 2 NFT collections and mint 10 NFTs for each collection
+	nftID1 := createNftHelper(t, &env)
+
+	mint, err := NewTokenMintTransaction().SetTokenID(nftID1).SetMetadatas(initialMetadataList).Execute(env.Client)
```

Comment:
nit: Just a note here `SetMetadatas` looks like bad naming. Haven't seen that up until now.

Reply:
That's how it was implemented in the past. In other sdks its's `SetMetadata` so we probably should make it the same.